### PR TITLE
adding all primitive to the supported types

### PR DESCRIPTION
**Pull request by jpnjfk**
The toString() in utils.go supports just nil, string, int and float64.
When a variable in the given map is explecitly defined with a type such as int64 and 
int32, the toString returns the error as "toString: Unsupported type: int64".
Here is an example code.

var v int64
data := map[string]interface{}{
    "value": v,
}
result, err := toString(data, "value")

So, all of the primitive types should appear in the switch-case block in the toString().
In addition, I'd like the function to support time.Duration.

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -32,7 +33,7 @@ func whitespace(start, end int, ptr_input *[]rune) (rune, int) {
 
 // toString retrieves a value from the given map and tries to return a string representation.
 //
-// It will returns an error if the value's type is not <nil/string/int/float64>.
+// It will returns an error if the value's type is not <premitive or time.Duration>.
```
Comment (gotnospirit) [inline utils.go]: We're implementing here a subset of numeric types so we can't pretend we did it for _all_ primitive types; therefore I would rephrase it as "<nil/string/numeric/time.Duration>".
Plus, to fully implement all the numeric types, we should add/extend cases for the missing ones: uintptr, complex64, complex128, byte & rune.

File: utils.go [hunk 2]
```diff
@@ -50,6 +51,39 @@ func toString(data map[string]interface{}, key string) (string, error) {
...
+			return strconv.FormatUint(uint64(v.(uint16)), 10), nil
+
+		case uint32:
+			return strconv.FormatUint(uint64(v.(uint32)), 10), nil
+
+		case uint64:
+			return strconv.FormatUint(v.(uint64), 10), nil
+
+		case time.Duration:
+			return v.(time.Duration).String(), nil
```
Comment (gotnospirit) [inline utils.go:95]: just on the form, could you rewrite it to follow this order:
- int, int8, int16, int32, int64
- uint, uint8, uint16, uint32, uint64
- float32, float64
- time.Duration

it just seems more logic to me, ty

File: utils_test.go [hunk 3]
```diff
@@ -91,3 +92,53 @@ func TestToString(t *testing.T) {
 	toStringResult(t, data, "I", "42")
 	toStringResult(t, data, "F", "0.305")
 }
+
+func TestToStringAdditionalTypes(t *testing.T) {
```
Comment (gotnospirit) [inline utils_test.go]: I think we should add specific tests for explicit numeric types and time duration. Something like TestToStringNumericTypes & TestToStringTimeDuration

File: utils_test.go [hunk 4]
```diff
@@ -91,3 +92,53 @@ func TestToString(t *testing.T) {
...
+	int16Max = 32767
+	uint32Max = 4294967295
+	int32Max = 2147483647
+	uint64Max = 18446744073709551615
+	int64Max = 9223372036854775807
+	float32Value = 3.14
+	du := time.Date(1970, 0, 1, 0, 0, 0, 0, time.UTC).Sub(time.Date(1960, 0, 1, 0, 0, 0, 0, time.UTC))
+	data := map[string]interface{}{
+		"duration" : du,
+		"byteMax"   : byteMax,
```
Comment (gotnospirit) [inline utils_test.go]: for the sake of brevity we could just write `"byteMax": byte(255)` and then remove the associated var declaration & assignation. we could do the same for the other variables and save 21 lines :)

File: utils_test.go [hunk 5]
```diff
@@ -91,3 +92,53 @@ func TestToString(t *testing.T) {
...
+		"int8Min"   : -128,
+		"int8Max"   : int8Max,
+		"int16Min"  : -32768,
+		"int16Max"  : int16Max ,
+		"int32Min"  : -2147483648 ,
+		"int32Max"  : int32Max ,
+		"int64Min"  : -9223372036854775808,
+		"int64Max" : int64Max,
+		"float32" : float32Value,
+	}
```
Comment (gotnospirit) [inline utils_test.go:120]: we should add keys for explicitly declared int/uint/float64

File: utils_test.go [hunk 6]
```diff
@@ -91,3 +92,53 @@ func TestToString(t *testing.T) {
...
+
+	toStringResult(t, data, "float32", "3.14")
+	toStringResult(t, data, "duration", "87672h0m0s")
+	toStringResult(t, data, "int8Max", "127")
+	toStringResult(t, data, "byteMax", "255")
+	toStringResult(t, data, "uint8Max", "255")
+	toStringResult(t, data, "uint64Max", "18446744073709551615")
+	toStringResult(t, data, "int64Min", "-9223372036854775808")
+	toStringResult(t, data, "int64Max", "9223372036854775807")
+}
```
Comment (gotnospirit) [inline utils_test.go:152]: some keys are not tested :(

on the form, it seems more logic to me to see the keys and their tests in the same order, ty

**Review by gotnospirit — CHANGES_REQUESTED** [review]
Hi!
I'm pretty open to add more support for builtin types to toString().
Before I merge it, please consider my comments.
Thank you for your time :)

**Comment by gotnospirit** [comment]
and "go fmt" please ;)

**Comment by jpnjfk** [comment]
Thank you for quickly reviewing my changes and sending your comments.
I made the changes based on your suggestions.
The 'byte' is an alias of 'uint8' and the 'rune' is an alias of 'int32' so that I didn't add the case clause for them to the switch-case block in toString().
But I added them in the tests.

File: utils.go (hunk 2 above)
Comment (jpnjfk) [reply utils.go:95]: Done.

File: utils_test.go (hunk 5 above)
Comment (jpnjfk) [reply utils_test.go:120]: Done.

File: utils_test.go (hunk 6 above)
Comment (jpnjfk) [reply utils_test.go:152]: Done.

File: utils.go (hunk 1 above)
Comment (jpnjfk) [reply utils.go]: Done.

File: utils_test.go (hunk 3 above)
Comment (jpnjfk) [reply utils_test.go]: Done.

File: utils_test.go (hunk 4 above)
Comment (jpnjfk) [reply utils_test.go]: done.

File: utils.go (hunk 1 above)
Comment (jpnjfk) [reply utils.go]: I didn't add 'rune' and 'byte' to the switch-case block in toPrint() because it's aliases of uint8, int32 respectively.
I did add them in the tests.

File: utils.go [hunk 7]
```diff
@@ -82,6 +76,21 @@ func toString(data map[string]interface{}, key string) (string, error) {
 		case uint64:
 			return strconv.FormatUint(v.(uint64), 10), nil
 
+		case float64:
+			return strconv.FormatFloat(v.(float64), 'f', -1, 64), nil
+
+		case float32:
+			return strconv.FormatFloat(float64(v.(float32)), 'f', -1, 32), nil
```
Comment (gotnospirit) [inline utils.go:80]: Could you reorder float32 then float64 please?

File: utils_test.go [hunk 8]
```diff
@@ -93,52 +93,60 @@ func TestToString(t *testing.T) {
...
+		"int8Min":    int8(-128),
+		"int8Max":    int8(127),
+		"int16Min":   int16(-32768),
+		"int16Max":   int16(32767),
+		"int32Min":   int32(-2147483648),
+		"int32Max":   int32(2147483647),
+		"int64Min":   int64(-9223372036854775808),
+		"int64Max":   int64(9223372036854775807),
+		"float32":    float32(3.14),
+		"float64":    float32(3.14e-10),
```
Comment (gotnospirit) [inline utils_test.go]: I think it should be `float64(3.14e-10)`

**Review by gotnospirit — CHANGES_REQUESTED** [review]
Thank you for the changes you added.
2 last comments and I would appreciate that you squash the commits you have done before merging.

File: utils.go (hunk 7 above)
Comment (jpnjfk) [reply utils.go:80]: done.

File: utils_test.go (hunk 8 above)
Comment (jpnjfk) [reply utils_test.go]: done.

**Comment by jpnjfk** [comment]
Thank you for reviewing my changes again. :)
I updated the points you suggested.

**Review by jpnjfk — COMMENTED** [review]
Thank you for reviewing my changes again. :)
I updated the points you suggested.

**Comment by jpnjfk** [comment]
Also, I squashed my commits to one commit.

**Comment by gotnospirit** [comment]
Thank you!

**Comment by jpnjfk** [comment]
Great!!!
Thank you for your help. :)
